api/proto/common/common.proto

```diff
@@ -11,19 +13,24 @@ message G1Commitment {
 }
 
 // BlobCommitment represents commitment of a specific blob, containing its
-// KZG commitment, degree proof, the actual degree, and data length in number of symbols.
+// KZG commitment, degree proof, the actual degree, and data length in number of symbols (field elements).
+// It deserializes into https://github.com/Layr-Labs/eigenda/blob/ce89dab18d2f8f55004002e17dd3a18529277845/encoding/data.go#L27
+//
+// See https://github.com/Layr-Labs/eigenda/blob/master/docs/spec/attestation/encoding.md#validation-via-kzg
+// to understand how this commitment is used to validate the blob.
 message BlobCommitment {
-  // A commitment to the blob data.
+  // Concatenation of the x and y coordinates of `common.G1Commitment`.
   bytes commitment = 1;
   // A commitment to the blob data with G2 SRS, used to work with length_proof
-  // such that the claimed length below is verifiable.
+	// such that the claimed length below is verifiable.
   bytes length_commitment = 2;
   // A proof that the degree of the polynomial used to generate the blob commitment is valid.
-  // It is computed such that the coefficient of the polynomial is committing with the G2 SRS
-  // at the end of the highest order.
+	// It is computed such that the coefficient of the polynomial is committing with the G2 SRS
+	// at the end of the highest order.
   bytes length_proof = 3;
-  // The length specifies the degree of the polynomial used to generate the blob commitment. The length
-  // must equal to the degree + 1, and it must be a power of 2.
+  // The length of the blob in symbols (field elements), which must be a power of 2.
```

Comment:
This is the one that I still have question about. I think it's just a upper bound of length/degree, rather than the length/degree of the poly.

Reply:
Not sure for v1, but for v2 I think its exactly the length now? cc @litt3

Reply:
I suppose you could call it an "upper bound", in the sense that you have the option choose to a power-of-2 length arbitrarily higher than the necessary length, without failing verification.

But I agree with @samlaf, we've been referring to this length as the source-of-truth, *actual length* of the blob. If the commitment creator chose a larger value than necessary, that means they essentially just appended a bunch of 0s to the blob polynomial, and therefore literally increased the length of the blob (even though they didn't add any additional non-0 information)

Reply:
But the appended 0s even though not transmitted from clients to disperser, will actually be encoded as chunks and transmitted to operators, which is why we charge for them right? So in this sense the blob_length is LITERALLY the blob_length, as seen from the network's perspective, right?

Reply:
Yes, that matches my understanding.

Reply:
That makes sense to me.  I think the source of confusion might be the word "upper bound"; if we just say it's any power of 2 that's larger than raw/original blob, and this power of 2 is the *actual* blob viewed by network, that's clear to me.
Anyway, at least there is no use of "upper bound" in documentation here.